### PR TITLE
cluster: Lower sync config log level

### LINKED ISSUE #4923: Too many sync config logs if some store are down
## Bug Report

There are too many sync config logs.
```
[2022/05/10 18:06:33.371 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:07:32.902 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:07:33.070 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:07:33.199 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:07:43.723 +00:00] [INFO] [cluster_worker.go:131] ["alloc ids for region split"] [region-id=135118037] [peer-ids="[135118038,135118039,135118040]"]
[2022/05/10 18:07:43.725 +00:00] [INFO] [region.go:567] ["region Version changed"] [region-id=15500] [detail="StartKey Changed:{7480000000000000FF4E5F728000000014FF0F25180000000000FA} -> {7480000000000000
FF4E5F728000000014FF17563F0000000000FA}, EndKey:{7480000000000000FF4E5F728000000014FF1BB9FF0000000000FA}"] [old-version=1742] [new-version=1743]
[2022/05/10 18:07:43.725 +00:00] [INFO] [cluster_worker.go:223] ["region batch split, generate new regions"] [region-id=15500] [origin="id:135118037 start_key:\"7480000000000000FF4E5F728000000014FF0F25180
000000000FA\" end_key:\"7480000000000000FF4E5F728000000014FF17563F0000000000FA\" region_epoch:<conf_ver:29 version:1743 > peers:<id:135118038 store_id:4 > peers:<id:135118039 store_id:5 > peers:<id:135118
040 store_id:1 >"] [total=1]
[2022/05/10 18:08:32.902 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:08:33.069 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:08:33.182 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:09:18.219 +00:00] [INFO] [cluster_worker.go:131] ["alloc ids for region split"] [region-id=135118041] [peer-ids="[135118042,135118043,135118044]"]
[2022/05/10 18:09:18.223 +00:00] [INFO] [region.go:567] ["region Version changed"] [region-id=1535] [detail="StartKey Changed:{7480000000000000FF4A5F728000000000FFEAD5020000000000FA} -> {7480000000000000F
F4A5F728000000000FFED38590000000000FA}, EndKey:{7480000000000000FF4A5F728000000000FFEE727E0000000000FA}"] [old-version=417] [new-version=418]
[2022/05/10 18:09:18.223 +00:00] [INFO] [cluster_worker.go:223] ["region batch split, generate new regions"] [region-id=1535] [origin="id:135118041 start_key:\"7480000000000000FF4A5F728000000000FFEAD50200
00000000FA\" end_key:\"7480000000000000FF4A5F728000000000FFED38590000000000FA\" region_epoch:<conf_ver:53 version:418 > peers:<id:135118042 store_id:1 > peers:<id:135118043 store_id:5 > peers:<id:13511804
4 store_id:4 >"] [total=1]
[2022/05/10 18:09:33.209 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:09:33.378 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:09:33.548 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:10:32.928 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:10:33.095 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:10:33.341 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
[2022/05/10 18:11:32.955 +00:00] [WARN] [cluster.go:322] ["sync store config failed, it will try next store"] [error="Get \"http://0.0.0.0:20180/config\": dial tcp 0.0.0.0:20180: connect: connection refus
ed"]
```

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?

Issue Number: Close #4923

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

Code changes

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]
LGTM, but how do we know it indeed synchronizes successfully?

**Comment by bufferflies** [comment]
> LGTM, but how do we know it indeed synchronizes successfully?

In store manager,  if store config is different , some log will be printed: https://github.com/tikv/pd/blob/6554d63c7350f9bcbe2f5ff50d81808398aade67/server/config/store_config.go#L162-L173

**Comment by rleungx** [comment]
> > LGTM, but how do we know it indeed synchronizes successfully?
> 
> In store manager, if store config is different , some log will be printed:
> 
> https://github.com/tikv/pd/blob/6554d63c7350f9bcbe2f5ff50d81808398aade67/server/config/store_config.go#L162-L173

I mean that we don't know if the config is the same or the synchronization just failed.

**Comment by bufferflies** [comment]
> > > LGTM, but how do we know it indeed synchronizes successfully?
> > 
> > 
> > In store manager, if store config is different , some log will be printed:
> > https://github.com/tikv/pd/blob/6554d63c7350f9bcbe2f5ff50d81808398aade67/server/config/store_config.go#L162-L173
> 
> I mean that we don't know if the config is the same or the synchronization just failed.

how about add one metric?

File: server/cluster/metrics.go [hunk 1]
```diff
@@ -96,6 +96,14 @@ var (
 			Help:      "The current progress of corresponding action",
 		}, []string{"address", "store", "action"})
 
+	storeSyncCount = prometheus.NewGaugeVec(
```
Comment (rleungx) [inline server/cluster/metrics.go:99]: the counter is enough?

File: server/cluster/metrics.go [hunk 2]
```diff
@@ -96,6 +96,14 @@ var (
 			Help:      "The current progress of corresponding action",
 		}, []string{"address", "store", "action"})
 
+	storeSyncCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "pd",
+			Subsystem: "cluster",
+			Name:      "store_sync",
+			Help:      "The state of store sync",
```
Comment (lhy1024) [inline server/cluster/metrics.go:104]:
```suggestion
			Help:      "The state of store sync config",
```

File: server/cluster/metrics.go [hunk 3]
```diff
@@ -96,6 +96,14 @@ var (
 			Help:      "The current progress of corresponding action",
 		}, []string{"address", "store", "action"})
 
+	storeSyncCount = prometheus.NewCounterVec(
```
Comment (lhy1024) [inline server/cluster/metrics.go:99]: how about `storeSyncConfigEvent`

File: server/cluster/metrics.go (hunk 2 above)
Comment (bufferflies) [reply server/cluster/metrics.go]: fix

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**